packages/components/src/select-control/index.js

```diff
@@ -92,21 +93,29 @@ export class SelectControl extends Component {
 	}
 
 	selectOption( option ) {
-		const { multiple, onChange, selected } = this.props;
-		const { query } = this.state;
+		const { multiple, selected } = this.props;
 		const newSelected = multiple ? [ ...selected, option ] : [ option ];
 
 		this.reset( newSelected );
 
+		const oldSelected = Array.isArray( selected )
+			? selected
+			: [ { key: selected } ];
+		const isSelected = findIndex( oldSelected, { key: option.key } );
+		if ( isSelected === -1 ) {
+			this.setNewValue( newSelected );
+		}
+	}
+
+	setNewValue( newValue ) {
+		const { onChange, selected, multiple } = this.props;
+		const { query } = this.state;
 		// Trigger a change if the selected value is different and pass back
 		// an array or string depending on the original value.
-		if ( Array.isArray( selected ) ) {
-			const isSelected = findIndex( selected, { key: option.key } );
-			if ( isSelected === -1 ) {
-				onChange( newSelected, query );
-			}
-		} else if ( selected !== option.key ) {
-			onChange( option.key, query );
+		if ( multiple || Array.isArray( selected ) ) {
+			onChange( newValue, query );
+		} else if ( newValue.length < 2 ) {
```

Comment:
Could you explain this bit of logic?

Reply:
Yeah I should probably change that back to an `else` as it is kinda redundant.
It is suppose to check that there is only max 1 item selected, but in this case if that doesn't happen it will never trigger `onChange`, which is also not a great solution.
Will stick with an `else`.